### PR TITLE
Add JNI tests

### PR DESCRIPTION
**Pull request by NigelYiboYu**
Add 6 types of JNI tests to measure JNI call overhead.

Signed-off-by: Nigel Yu <yunigel@ca.ibm.com>

### CODE REVIEW

File: JNI-test-scripts/linux/build-dll.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+cd ../../
+HOME_DIR=`pwd`
+
+echo Using $HOME_DIR as bumblebench home directory.
+
+rm -f ./*.o
+rm -f ./*.so
+
+
+CPP_DIR="$HOME_DIR/com/ibm/bumblebench/jni/c++/"
```
Comment (smlambert) [inline JNI-test-scripts/linux/build-dll.sh]: net/adoptopenjdk/bumblebench

File: JNI-test-scripts/linux/build-dll.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
+
```
Comment (smlambert) [inline JNI-test-scripts/linux/build-dll.sh:14]: For new files, we add the Apache v2 license at the top of them for the AdoptOpenJDK projects (see https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/maketest.sh#L2 for example of header for shell script), applies to all new files in this PR

File: net/adoptopenjdk/bumblebench/jni/ArrayReadWriteRegionBench.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package net.adoptopenjdk.bumblebench.jni;
+
```
Comment (smlambert) [inline net/adoptopenjdk/bumblebench/jni/ArrayReadWriteRegionBench.java:15]: Please also add Apache v2 license header to all new Java files, see [another java file for example](https://github.com/AdoptOpenJDK/bumblebench/blob/master/net/adoptopenjdk/bumblebench/core/BumbleBench.java#L1) - applies to all new files in this PR

File: net/adoptopenjdk/bumblebench/jni/c++/CallOverheadTestcases.cpp [hunk 4]
```diff
@@ -0,0 +1,73 @@
+#include <jni.h>
```
Comment (smlambert) [inline net/adoptopenjdk/bumblebench/jni/c++/CallOverheadTestcases.cpp:14]: This file also needs a apache v2 header

File: net/adoptopenjdk/bumblebench/jni/ArrayReadWriteRegionBench.java (hunk 3 above)
Comment (NigelYiboYu) [reply net/adoptopenjdk/bumblebench/jni/ArrayReadWriteRegionBench.java:15]: Apache header is added to all new files (Java and shell scripts).

File: JNI-test-scripts/linux/build-dll.sh (hunk 2 above)
Comment (NigelYiboYu) [reply JNI-test-scripts/linux/build-dll.sh:14]: Fixed

File: JNI-test-scripts/linux/build-dll.sh (hunk 1 above)
Comment (NigelYiboYu) [reply JNI-test-scripts/linux/build-dll.sh]: Fixed

File: net/adoptopenjdk/bumblebench/jni/c++/CallOverheadTestcases.cpp (hunk 4 above)
Comment (NigelYiboYu) [reply net/adoptopenjdk/bumblebench/jni/c++/CallOverheadTestcases.cpp:14]: Fixed in all cpp files

**Review by smlambert — APPROVED** [review]
Thanks @NigelYiboYu !

**Pull request merged**